packages/aws-cdk-lib/verify-stripped-exp.js

```diff
@@ -0,0 +1,134 @@
+// +------------------------------------------------------------------------------------------------
+// | this script runs post pack to verify that experimental modules in aws-cdk-lib includes **only**  L1 autogenerated files. 
+// | The purpose is to avoid publishing L2 of experimental modules with aws-cdk-lib
+// |
+const { spawnSync } = require('child_process');
+const os = require('os');
+const fs = require('fs-extra');
+const path = require('path');
+
+
+const JS_DIST_LIB = 'dist/js/';
+
+async function main(tempDir) {
+  const cwd = process.cwd();
+  const awsCdkModulesRepoPath = path.join(findWorkspacePath(),'packages', '@aws-cdk');
+  process.chdir(tempDir);
+  const version = require('./package.json').version;
+  const tarFullPath = path.join(cwd, JS_DIST_LIB, `aws-cdk-lib-${version}.tgz`);
+
+  // install the tarball in a temp directory
+  exec('npm', ['init', '-y']);
+  exec('npm', ['install', tarFullPath]);
+  const installedAwsCdkLibPath = path.join('node_modules', 'aws-cdk-lib', 'lib');
+
+  for (const module of fs.readdirSync(awsCdkModulesRepoPath)) {
+    const pkgJson = require(path.join(awsCdkModulesRepoPath, module, 'package.json'));
+    if (pkgJson['stability'] !== 'experimental') {
+      continue;
+    }
+    if (pkgJson['cdk-build'].cloudformation) {
+      // if a cfn module, verify only the allowed files exists
+      const files = await getAllFiles(path.join(installedAwsCdkLibPath, module));
+      files.forEach(file => {
+        if (!isAllowedFile(file)) {
+          throw `only L1 auto generated files are allowed in experimental modules included in aws-cdk-lib, the file ${file} exists in ${module}`;
+        }
+      });
```

Comment:
Better to accumulate the disallowed files and throw an exception with all of them, to make it faster to fix.

Reply:
finnnneeee. Added pretty print just for you :)